src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataScriptTempFileStream.cs

```diff
@@ -0,0 +1,117 @@
+        {
+            var tempFileName = $"{serverName}.tmp";
+            var generatedScripts = scripts.ToList();
+
+            try
+            {
+                var tempFilePath = Path.Combine(Path.GetTempPath(), tempFileName);
+                using (StreamWriter sw = new StreamWriter(tempFilePath, false))
+                {
+                    foreach (var script in generatedScripts)
+                    {
+                        sw.WriteLine(script);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.Warning($"Failed to write scripts to temporary file. Error: {ex.Message}");
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Reads the scripts associated with the provided server name.
+        /// </summary>
+        /// <param name="serverName">The name of the server to retrieve the scripts for.</param>
+        /// <returns>List containing all the scripts in the file.</returns>
+        public static IEnumerable<string> Read(string serverName)
+        {
+            var tempFileName = $"{serverName}.tmp";
+            var scripts = new List<string>();
+
+            try
+            {
+                var tempFilePath = Path.Combine(Path.GetTempPath(), tempFileName);
+                if (!File.Exists(tempFilePath))
+                {
+                    return scripts;
+                }
+
+                using (StreamReader sr = new StreamReader(tempFilePath))
+                {
+                    while (!sr.EndOfStream)
+                    {
+                        var line = sr.ReadLine();
+                        if (!String.IsNullOrWhiteSpace(line))
+                        {
+                            scripts.Add(line);
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.Warning($"Failed to read scripts from temporary file. Error: {ex.Message}");
+                throw;
+            }
+
+            return scripts;
+        }
+
+        /// <summary>
+        /// Checks to see if the temporary file containing the scripts for a server needs to be updated.
+        /// </summary>
+        /// <param name="serverName">The name of the file associated with the given server name.</param>
+        /// <returns>Flag indicating that the script file is valid.</returns>
+        public static bool IsScriptTempFileUpdateNeeded(string serverName)
+        {
+            var tempFileName = $"{serverName}.tmp";
+
+            try
+            {
+                var tempFilePath = Path.Combine(Path.GetTempPath(), tempFileName);
+                if (!File.Exists(tempFilePath))
+                {
+                    return false;
+                }
+                else
+                {
+                    var lastWriteTime = File.GetLastWriteTime(tempFilePath);
```

Comment:
Would be good to put a comment here on why you chose 30 days and any other info that would be useful for anyone trying to understand how this whole thing works.